backend/hitas/models/company.py

```diff
@@ -0,0 +1,129 @@
+        GREATER_THAN_30_YEARS_PLOT_DEPARTMENT_NOTIFICATION = _("Greater than 30 years, plot department notification")
+        READY_NO_STATISTICS = _("Ready, no statistics")
+
+
+class Company(models.Model):
+    uuid = models.UUIDField(default=uuid.uuid4, editable=False)
+
+    # Official spelling of the company name
+    official_name = models.CharField(max_length=1024)
+    # More human-friendly company name
+    display_name = models.CharField(max_length=1024)
+
+    state = EnumIntegerField(CompanyState, default=CompanyState.NOT_READY)
+    # Business ID / 'y-tunnus'
+    business_id = models.CharField(max_length=9, validators=[validate_business_id])
+
+    building_type = models.ForeignKey(BuildingType, on_delete=models.PROTECT)
+    financing_type = models.ForeignKey(FinancingMethod, on_delete=models.PROTECT)
+    public_officer = models.ForeignKey(PublicOfficer, on_delete=models.PROTECT)
+    property_manager = models.ForeignKey(PropertyManager, on_delete=models.PROTECT)
+    developer = models.ForeignKey(Developer, on_delete=models.PROTECT)
+
+    # 'hankinta-arvo'
+    acquisition_price = models.DecimalField(max_digits=15, decimal_places=2)
+    # 'toteutunut hankinta-arvo'
+    realized_acquisition_price = models.DecimalField(null=True, blank=True, max_digits=15, decimal_places=2)
+
+    # 'kiinteistötunnus'
+    property_identifier = models.CharField(max_length=19, validators=[validate_property_id])
+
+    legacy_id = models.CharField(max_length=10, null=True, blank=True)
+
+    addresses = models.ManyToManyField(Address, through='CompanyAddress')
+
+    class Meta:
+        verbose_name_plural = 'Companies'
+
+    def __str__(self):
+        return self.display_name
+
+
+# Rakennustunnus
+class BuildingIdentifier(models.Model):
+    value = models.CharField(max_length=1024)
+    company = models.ForeignKey(Company, on_delete=models.PROTECT)
+
+    def __str__(self):
+        return self.value
+
+
+class CompanyAddress(models.Model):
+    company = models.ForeignKey(Company, on_delete=models.CASCADE)
+    address = models.ForeignKey(Address, on_delete=models.CASCADE)
+    default_address = models.BooleanField(default=False)
+
+    def save(self, *args, **kwargs):
+        if not self.default_address:
+            return super(CompanyAddress, self).save(*args, **kwargs)
+
+        with transaction.atomic():
+            CompanyAddress.objects.filter(
+                company=self.company,
+                default_address=True,
+            ).update(default_address=False)
+        return super(CompanyAddress, self).save(*args, **kwargs)
+
+
+class CompanyAddressAdmin(admin.TabularInline):
+    model = Company.addresses.through
+    extra = 0
+
+
+class BuildingIdentifierAdmin(admin.TabularInline):
+    model = BuildingIdentifier
+    extra = 1
+
+
+@admin.register(Company)
+class CompanyAdmin(admin.ModelAdmin):
+    inlines = (CompanyAddressAdmin, BuildingIdentifierAdmin)
```

Comment:
Can you make your editor add newlines to end of file on save when it's missing, to not have Github warn about them? :)

Reply:
yep, seems the `format.sh` did not work for me for some reason